File: game/cave/scylla_spring.py
import pygame

from animation import Animation
from audio_helper import AudioHelper
from cave.cave_resources import CaveResources
from cave.family_cage_opens import FamilyCageOpens
from phone_events import PhoneEvents
from state import State


class ScyllaSpring(State):
    def process_events(self, phone_events: PhoneEvents):
        if self.get_frame_index() >= len(CaveResources.scylla_spring):
            return FamilyCageOpens

        if self.one_shot(1.5, "Fjeder"):
            AudioHelper.play(CaveResources.fjeder, self.context.audio_port)

        if self.one_shot(2, "AfskyliaSkydUd"):
            AudioHelper.play(CaveResources.afskylia_skyd_ud, self.context.audio_port)

        return None

    def render(self, screen):
        screen.blit(Animation.get_frame(CaveResources.scylla_spring, self.get_frame_index()), (0, 0))

File: game-web/src/cave/scyllaSpring.ts
/** Port of game/cave/scylla_spring.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { CaveResources } from './caveResources';
import { FamilyCageOpens } from './familyCageOpens';
import type { PhoneEvents } from '../phoneEvents';

export class ScyllaSpring extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    if (this.getFrameIndex() >= CaveResources.scyllaSpring.length) return FamilyCageOpens;

    if (this.oneShot(1.5, 'Fjeder'))        AudioHelper.play(CaveResources.fjeder);
    if (this.oneShot(2, 'AfskyliaSkydUd')) AudioHelper.play(CaveResources.afskyliaSkyudUd);

    return null;
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(CaveResources.scyllaSpring, this.getFrameIndex()), 0, 0);
  }
}
